test/test_api.py

```diff
@@ -0,0 +1,220 @@
+"""These tests ensure the API of how this can be used is enforced."""
+import os
+import time
+
+from pynisher import EMPTY, Pynisher, contexts, limit, supports
+from pynisher.util import memconvert
+
+import pytest
+
+
+def subfunction() -> int:
+    """Small test function which returns the id"""
+    return os.getpid()
+
+
+def err_function() -> None:
+    """Function that raises an error"""
+    raise RuntimeError("Error from test func `err_function`")
+
+
+def sleepy(sleep: float) -> None:
+    """Sleeps for `sleep` second"""
+    start = time.perf_counter()
+    while True:
+        duration = time.perf_counter() - start
+        if duration > sleep:
+            break
+
+
+def usememory(x: int) -> int:
+    """Use a certain amount of memory in B"""
+    bytearray(int(x))
+    return x
+
+
+def return_none() -> None:
+    """Just returns None to make sure we pass it back correctly"""
+    return None
+
+
+@pytest.mark.parametrize("context", contexts)
+def test_as_contextmanager(context: str) -> None:
+    """
+    Expects
+    -------
+    * Should be able to use as a context manager
+    """
+    with Pynisher(subfunction, context=context) as restricted_func:
+        other_process_id = restricted_func()
+
+    this_process_id = os.getpid()
+    assert this_process_id != other_process_id
+
+
+@pytest.mark.parametrize("context", contexts)
+def test_call(context: str) -> None:
+    """
+    Expects
+    -------
+    * Should be able to call the restricted function
+    """
+    restricted_func = Pynisher(subfunction, context=context)
+
+    this_process_id = os.getpid()
+    other_process_id = restricted_func()
+
+    assert this_process_id != other_process_id
+
+
+@pytest.mark.parametrize("context", contexts)
+def test_run(context: str) -> None:
+    """
+    Expects
+    -------
+    * Should be able to explicitly call run on the restricted function
+    """
+    pynisher = Pynisher(subfunction, context=context)
+
+    this_process_id = os.getpid()
+    other_process_id = pynisher.run()
+    assert this_process_id != other_process_id
+
+
+def test_limit_usage() -> None:
```

Comment:
Why isn't there a context fixture on this one?

Reply:
It's not supported with spawn and only really supported on Linux

Reply:
Okay, so the context manager version can only be used on Linux?

Reply:
I updated the tests to what works.

Reply:
Sorry i misread which test this is. I added `contexts`. All systems work with the context manageer, I was confusing it with decorator.